src/NServiceBus.Core/Pipeline/Outgoing/OutgoingContext.cs

```diff
@@ -11,10 +11,14 @@ protected OutgoingContext(string messageId, Dictionary<string, string> headers,
             : base(parentContext)
         {
             Headers = headers;
-            MessageId = messageId;
+
+            if (!Headers.ContainsKey(NServiceBus.Headers.MessageId))
```

Comment:
I have know to conflicting opinions. Yours and mikes. I updated this PR to
move towards mikes. Let's get on a call and hash it out. I'm not willing to
let this drag on over days on the PR. Ok?

On Monday, July 25, 2016, Tim Bussmann <notifications@github.com> wrote:

> In src/NServiceBus.Core/Pipeline/Outgoing/OutgoingContext.cs
> <https://github.com/Particular/NServiceBus/pull/3961#discussion_r72070096>
> :
>
> > @@ -11,10 +11,14 @@ protected OutgoingContext(string messageId, Dictionary<string, string> headers,
> >              : base(parentContext)
> >          {
> >              Headers = headers;
> > -            MessageId = messageId;
> > +
> > +            if (!Headers.ContainsKey(NServiceBus.Headers.MessageId))
>
> not entirely convinced by this approach. Why not do this on
> ExtendableOptions or set the header when we create the OutgoingSendContext
> (+the related contexts)? In our discussion we agreed that keeping
> MessageId isn't even a requirement? I think I'm still on that side. If
> you decide to mess with headers, don't expect other properties to
> automatically update, we don't do that with other context properties.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/3961/files/7b7314607cd5902c5ffb9ba9ae336fbfd3840fd4#r72070096>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAKoslvNn2NtXTl-0T_XhTM6IFgsq6A6ks5qZMRsgaJpZM4JSxku>
> .
>